biojava-structure/src/main/java/org/biojava/nbio/structure/symmetry/core/PermutationGroup.java

```diff
@@ -142,16 +142,16 @@ public int hashCode() {
 
 	@Override
 	public boolean equals(Object obj) {
-	    if (this == obj) {
-		return true;
-	    }
-	    if (obj == null) {
-		return false;
-	    }
-	    if (this.getClass() == obj.getClass()) {
-		return permutations.equals(((PermutationGroup)obj).permutations);
-	    }
-	    return false;
-	}
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (this.getClass() == obj.getClass()) {
+            return getGroupTable().equals(((PermutationGroup) obj).getGroupTable());
+        }
+        return false;
+    }
```

Comment:
I think it's correct but unnecessary. `getGroupTable()` depends only on the `permutations` member and fully represents all information in it. I guess you could argue that using the `getGroupTable()` function in both `hashCode` and `equals` makes their correctness more obvious, but I'm not sure it's worth the additional computation to convert `permutations` into a string.